packages/playground/src/components/reserve_action_btn.vue

```diff
@@ -0,0 +1,149 @@
+<template>
+  <container>
+    <v-dialog v-model="openUnreserveDialog" max-width="600" class="custom-dialog">
+      <v-card>
+        <v-card-title class="pt-4">
+          <span style="display: flex; align-items: center">
+            <v-icon color="warning" style="margin-right: 8px">mdi-alert</v-icon>
+            Are you sure you want to unreserve this dedicated node?
+          </span>
+        </v-card-title>
+        <v-divider style="color: #1aa18f" />
+        <v-card-text>This will free up the node for others on the chain</v-card-text>
+        <v-card-actions class="justify-end">
+          <v-btn variant="outlined" color="primary" @click="unReserveNode" :loading="loadingUnreserveNode">Yes</v-btn>
+          <v-btn variant="outlined" color="primary" @click="openUnreserveDialog = false">No</v-btn>
+        </v-card-actions>
+      </v-card>
+    </v-dialog>
+    <v-btn
+      small
+      outlined
+      :loading="loadingReserveNode"
+      :disabled="disableButton"
+      color="#064663"
+      style="background: #1aa18f"
+      v-if="rentedByTwinId === 0"
+      @click="reserveNode"
+    >
+      Reserve
+    </v-btn>
+    <v-btn
+      small
+      outlined
+      color="red"
+      :loading="loadingUnreserveBtn"
+      :disabled="disableButton"
+      style="background: #1aa18f"
+      v-if="rentedByTwinId === twinId"
+      @click="removeReserve"
+    >
+      Unreserve
+    </v-btn>
+    <v-btn small outlined disabled color="gray" v-if="rentedByTwinId !== 0 && rentedByTwinId !== twinId"> Taken </v-btn>
+  </container>
+</template>
+
+<script lang="ts">
+import { ref } from "vue";
+
+import { useProfileManager } from "../stores";
+import { createCustomToast, ToastType } from "../utils/custom_toast";
+import { getGrid } from "../utils/grid";
+
+const profileManager = useProfileManager();
+
+export default {
+  name: "ReserveBtn",
+  props: {
+    rentedByTwinId: {
+      type: Number,
+      required: true,
+    },
+    nodeId: {
+      type: String,
+      required: true,
+    },
+    twinId: {
+      type: Number,
+      required: true,
+    },
```

Comment:
Why not pass the entire node object instead of extracting these values here? It would simplify the process and allow us to access these fields directly from the node object within this component.

also the twinId, can be loaded from the `profileManager`

```ts
profileManager.profile?.twinId"
```